hamp_radar/geometries.py

```diff
@@ -75,3 +77,49 @@ class MultiMainBlockGeometry:
     count: int
     step: Optional[int]
     subblocks: List[SingleSubBlockGeometry]
+
+
+@dataclass
+class DatasetBlockGeometry:
+    """
+    Representation of a block of a dataset is a multimainblock with
+    the name of the .pds file it's sourced from.
+
+    Attributes:
+        filename (Path): The name of the .pds file the mmbg is from.
+        mainblock (MultiMainBlockGeometry): The mmbg in the file.
+    """
+
+    filename: Path
+    mainblock: MultiMainBlockGeometry
+
+
+@dataclass
+class DatasetGeometry:
+    """
+    Representation of a dataset's geometry is a DatasetBlockGeometry instance
+    for the PPAR and a list of DatasetBlockGeometry instances
+    for the data associated with that DSP configuration.
+
+    Attributes:
+        ppar (DatasetBlockGeometry): The block for the PPAR in the file.
+        data (List[DatasetBlockGeometry]): The blocks for the data associated with that PPAR.
+    """
+
+    ppar: DatasetBlockGeometry
+    data: List[DatasetBlockGeometry]
+
+
+@serde
```

Comment:
(as in the other PR, I'd guess that this should work as well)
```suggestion
@dataclass
```

Reply:
snap again ;)